alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -400,6 +400,16 @@ class TransportIndexMonitorAction @Inject constructor(
         private suspend fun indexMonitor() {
             var metadata = createMetadata()
 
+            if (user != null) {
+                // Use the backend roles which is an intersection of the requested backend roles and the user's backend roles.
+                // Admins can pass in any backend role. Also if no backend role is passed in, all the user's backend roles are used.
+                val rbacRoles = if (request.rbacRoles.isNullOrEmpty())
+                    user.backendRoles.toSet()
+                else if (!isAdmin(user)) request.rbacRoles!!.intersect(user.backendRoles).toSet()
+                else request.rbacRoles!!
```

Comment:
I know we're doing an `isNullOrEmpty()` check for the `if` here but we should try to avoid `!!` in production as much as possible. If empty `request.rbacRoles` are not logically different in how they're evaluated vs `null`, we should just default the argument to an empty list instead and not make it nullable.

Reply:
I have fixed it